src/lib.rs

```diff
@@ -214,22 +370,103 @@ impl ErrorPoint {
     }
 }
 
+
+/// represent a key-value pair
+#[derive(Serialize, Debug, Clone)]
+pub struct KvPair {
+    key: &'static str,
+    value: ThrowContextValues,
+}
+
+impl KvPair {
+    fn new(key: &'static str, value: ThrowContextValues) -> KvPair {
+        KvPair { key, value }
+    }
+    ///key
+    pub fn key(&self) -> &'static str {
+        self.key
+    }
+    ///value
+    pub fn value(&self) -> &ThrowContextValues {
+        &self.value
+    }
+}
+
+
 /// Represents an error. Stores an original error of type E, and any number of ErrorPoints at
 /// which the error was propagated.
 pub struct Error<E> {
     points: Vec<ErrorPoint>,
+    context: Vec<KvPair>,
     error: E,
 }
 
+struct VecErrorPoint<'a>(&'a [ErrorPoint]);
+
+impl<'a> Serialize for VecErrorPoint<'a>
+{
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
+        for e in self.0 {
+            seq.serialize_element(e)?;
+        }
+        seq.end()
+    }
+}
+
+struct VecKvPair<'a>(&'a [KvPair]);
+
+impl<'a> Serialize for VecKvPair<'a>
+{
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        let mut seq = serializer.serialize_seq(Some(self.0.len()))?;
```

Comment:
Would it make sense to just call `self.0.serialize(serializer)` here to defer to `[]`'s serialization completely?

Reply:
done